internal/users/localentries/localgroups.go

```diff
@@ -114,142 +140,167 @@ func existingLocalGroups(user, groupPath string) (groups []string, err error) {
 		if t == "" {
 			continue
 		}
-		elems := strings.Split(t, ":")
-		if len(elems) != 4 {
-			return nil, fmt.Errorf("malformed entry in group file (should have 4 separators): %q", t)
+		elems := strings.SplitN(t, ":", 4)
+		if len(elems) < 4 {
+			return nil, fmt.Errorf("malformed entry in group file (should have 4 separators, got %d): %q", len(elems), t)
 		}
 
-		n := elems[0]
-		users := strings.Split(elems[3], ",")
-		if !slices.Contains(users, user) {
-			continue
+		name, passwd, gidValue, usersValue := elems[0], elems[1], elems[2], elems[3]
+
+		gid, err := strconv.ParseUint(gidValue, 10, 0)
+		if err != nil || gid > math.MaxUint32 {
+			return nil, fmt.Errorf("failed parsing entry %q, unexpected GID value", t)
```

Comment:
Same here, let's not return an error if we encounter an invalid line

Reply:
Well, in the overflow case what we do though?
As setting it to 0 would be wrong (even if we don't validate it later, but then we would not write back as it is), even worse with `math.MaxUint32`... So the other option is keeping the invalid items tracked as strings, but then the rest is a bit annoying.

So... Not sure, TBH this is one of those cases in which it's acceptable to fail or skip the item completely IMHO (or we keep is as a string of invalid entries that we rewrite as they are, but without preserving the order).

Reply:
Yeah, I didn't check but doing something similar, in regards to keeping the old values (see https://github.com/ubuntu/authd/pull/993/commits/755d5d7f17587ac79a57ef66cd434cff3a766eb8)